File: VB/MyAdvBandedGridView/MyAdvBandedGridInfoRegistrator.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.Handler
Imports DevExpress.XtraGrid.Views.Base.ViewInfo

Namespace MyXtraGrid
	Public Class MyAdvBandedGridInfoRegistrator
		Inherits AdvBandedGridInfoRegistrator
		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyAdvBandedGridView"
			End Get
		End Property
		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New MyAdvBandedGridView(TryCast(grid, GridControl))
		End Function
		Public Overrides Function CreateViewInfo(ByVal view As BaseView) As BaseViewInfo
			Return New MyAdvBandedGridViewInfo(TryCast(view, MyAdvBandedGridView))
		End Function
		Public Overrides Function CreateHandler(ByVal view As BaseView) As BaseViewHandler
			Return New MyAdvBandedGridViewHandler(TryCast(view, MyAdvBandedGridView))
		End Function
	End Class
End Namespace

File: CS/MyAdvBandedGridView/MyAdvBandedGridInfoRegistrator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.Handler;
using DevExpress.XtraGrid.Views.Base.ViewInfo;

namespace MyXtraGrid
{
    public class MyAdvBandedGridInfoRegistrator : AdvBandedGridInfoRegistrator
    {
        public override string ViewName { get { return "MyAdvBandedGridView"; } }
        public override BaseView CreateView(GridControl grid) { return new MyAdvBandedGridView(grid as GridControl); }
        public override BaseViewInfo CreateViewInfo(BaseView view) { return new MyAdvBandedGridViewInfo(view as MyAdvBandedGridView); }
        public override BaseViewHandler CreateHandler(BaseView view) { return new MyAdvBandedGridViewHandler(view as MyAdvBandedGridView); }
    }
}
